### ISSUE TITLE
Add tray icon for AnyType process running in background

### ISSUE THREAD

**Issue by hsandt** [anyproto/anytype-ts#102]
### Have you read a contributing guide?

- [X] I have read CONTRIBUTING.md
- [X] I have searched the existing requests and didn't find any that were similar
- [X] I have considered creating a pull request instead and want to proceed

### Clear and concise description of the problem

Note: link to CONTRIBUTING in the issue template (text you see at the top when adding new Feature request) (https://github.com/anyproto/docs/CONTRIBUTING.md) is dead. Also, hyperlink has typo "CONTRINUTING.md"

After closing AnyType with Linux Ubuntu Unity desktop launcher > right-click on app > Quit or closing the window, I noticed that the app keeps running in the background. This is not an issue specific to this app, others like Discord have exactly the same issue. This is not what my issue is about.

However, since there is no tray icon for AnyType running in the background, it is also difficult to spot until you open the System Monitor. In my case it was 2nd highest CPU usage after Chrome, sometimes 1st, because of graphical effects, so I detected it later when trying to release CPU usage.

### Suggested solution

Add a tray icon to indicate running process, that would show on as many platforms as possible. User could also click on it to stop process (true Quit).

I haven't tested other platforms so maybe it's already implemented on Windows and macOS.

I know that the Unity desktop is getting older on Ubuntu and sometimes tray icons have some issues, so if this is not supported but a tray icon displays correctly on Gnome desktop, I will understand.

### Alternative

As a user, make sure to quit app with Ctrl+Q, not Dash > right-click > Quit (at least on Unity desktop).

### Additional context

The nice graphical effects on the welcome phase are responsible for higher CPU usage (note entry with `anytype --gpu-preferences` in the screenshot below):

![image](https://github.com/anyproto/anytype-ts/assets/1067772/49df5512-7f01-4893-bada-4e5b594fd928)

I expect that outside this, AnyType won't be using as much CPU and it won't be that much of an issue.

**Comment by fuksman** [comment]
@hsandt thank you for the heads-up about the link; I will fix it.

**Comment by isle9** [comment]
This seems like it's a problem with the Unity desktop environment, since the system tray icons work in GNOME.

**Comment by KaKi87** [comment]
Hi,

I confirm there is no tray icon on Ubuntu Unity.

When trying Anytype the first time, I thought it was closed when closing the window, so when trying to open it again from the launcher, it wouldn't launch, then it's only when opening the system monitor that I noticed the background process was the reason why.

Thanks

**Comment by filip-anytype** [comment]
Hey @KaKi87. This is happening because of [this](https://github.com/anyproto/anytype-ts/issues/630#issue-2204918415) issue. Usually the app is supposed to re-open if you try running it again.

**Comment by KaKi87** [comment]
Except I would be able to reopen from tray if there was a tray icon, which is what the current issue is for. Thanks

**Comment by filip-anytype** [comment]
> When trying Anytype the first time, I thought it was closed when closing the window, so when trying to open it again from the launcher, it wouldn't launch, then it's only when opening the system monitor that I noticed the background process was the reason why.

@KaKi87 I was referring to this part.

Regarding the tray icons, these are supported on other platforms including Linux (both Gnome and KDE have working system tray icons). Do you only have issues with Anytype when it comes to system tray or are other apps affected as well?

**Comment by KaKi87** [comment]
> I was referring to this part.

Yes, I detailed what I experience to illustrate why a notification saying "AnyType is still running in the background" would have saved me time to figure out what the actual issue was, which is the missing tray icon. Please add that notification.

> Do you only have issues with Anytype when it comes to system tray or are other apps affected as well?

Actually, I knew all Electron apps are affected, but didn't realize AnyType was an Electron app.

Related issue : electron/electron#38979

**Comment by adhadse** [comment]
Add to this, on vanilla GNOME, we don't support the legacy tray-icon API like ubuntu does (and there isn't any work done yet). It's long been deprecated for a better alternative that was never created. So best bet is, if it does happen, we'll be required to ask all GNOME users to use some sort of extension for this.

[Status Icons and GNOME
](https://blogs.gnome.org/aday/2017/08/31/status-icons-and-gnome/)

**Comment by ra3orblade** [comment]
Closing as non-relevant. Feel free to reopen as you think otherwise.

**Comment by KaKi87** [comment]
We all do think otherwise, we explained everything already.

**Comment by sleroq** [comment]
@KaKi87 

As far as I can tell, this is an electron on Unity issue, and there's a known env var workaround.

What would you actually want from Anytype here?

- just document limitation and workaround
- a "Anytype is still running in the background" notification on close (personally wouldn't like to see this every time)

or something else?

**Comment by KaKi87** [comment]
> there's a known env var workaround

What's that ?

> a "Anytype is still running in the background" notification on close

Yes please.

> personally wouldn't like to see this every time

Of course, it should be disable-able in settings.

**Comment by sleroq** [comment]
@KaKi87 

> What's that ?

From the issue you linked:
`XDG_CURRENT_DESKTOP='' electron-app`

https://github.com/electron/electron/issues/38979#issuecomment-1630025534

I am on linux as well and anytype is visible in the tray.

**Comment by KaKi87** [comment]
How about automatically applying that workaround when necessary then ?
